### PR TITLE
Updated to 1.17

### PR DESCRIPTION
**Pull request by LordDeatHunter**

### CODE REVIEW

**Comment by LordDeatHunter** [comment]
This works fine in dev, but you might wanna give the mod dependencies another look.
It appears that REI is now requiring architectury.

File: src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java
Comment (zabi94) [inline src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java]: Is there any reason for this change? Because it currently breaks other stuff I'm working with

File: src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java
Comment (zabi94) [inline src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java:44]: default visibility triggers warnings on my IDE, and it is generally better to keep access to a minimum. This line should be reverted

File: src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java
Comment (zabi94) [inline src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java]: Same as other comment

File: src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java
Comment (zabi94) [inline src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java:61]: Same as other comment

File: src/main/java/zabi/minecraft/nbttooltip/parse_engine/BWHumanReadableParser.java
Comment (zabi94) [inline src/main/java/zabi/minecraft/nbttooltip/parse_engine/BWHumanReadableParser.java]: String concat chains allocate more objects than necessary. Since this might deal with a lot of strings very often, StringBuilders are the better choice here

**Review by zabi94 — CHANGES_REQUESTED** [review]
Thanks for the help! I'll need the things I commented in the review changed before I can actually merge though

File: src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java
Comment (LordDeatHunter) [reply src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java:44]: IntelliJ was complaining about it on my end, I can revert it as long as it doesn't mess anything up.

File: src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java
Comment (LordDeatHunter) [reply src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java]: image
Should I still change it back?

File: src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java
Comment (LordDeatHunter) [reply src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java:44]: image
This is in java 16, idk if that makes any differences.

Comment (zabi94) [reply src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java:44]: I guess I can just add an ignore annotation. Keep this change, it's ok

File: src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java
Comment (zabi94) [reply src/main/java/zabi/minecraft/nbttooltip/config/ConfigInstance.java]: This one yes, I need this reverted, I use the "static" keyword as a marker in another project of mine and in need it to be explicit in this case

File: build.gradle
Comment (zabi94) [inline build.gradle]: Oh, one last change before I'll download and build. 
I'd rather have the official maven repos, not cursemaven.
Please revert to the previous format and use the gradle.properties file to choose the correct version for dependencies

**Review by zabi94 — CHANGES_REQUESTED** [review]

File: build.gradle
Comment (LordDeatHunter) [reply build.gradle]: This is the issue I mentioned in the original message. I'm not sure if they're all updated on the maven, and I'm unsure how the new REI interacts with architectury.

Comment (zabi94) [reply build.gradle]: I think gradle takes care of any inherited dependency, that should not be a problem. As for REI, I think they moved to another repo, so newer versions will be there. I tried in my other mod, extra alchemy, and it downloaded fine, if you give me a couple minutes I'll find the correct link

Comment (zabi94) [reply build.gradle]: actually, the repo is still the same, the project however changed id, and was split in api and implementation. The new dependency code block is 
```gradle

modCompileOnly("me.shedaniel:RoughlyEnoughItems-api-fabric:${project.rei_version}") {
	exclude module: 'fabric-api'
}

modRuntime("me.shedaniel:RoughlyEnoughItems-fabric:${project.rei_version}") {
	exclude module: 'fabric-api'
}
```

**Comment by zabi94** [comment]
Sweet. I'll merge and test tomorrow when I'll get back from work

**Comment by zabi94** [comment]
Thanks again

**Comment by LordDeatHunter** [comment]
Is everything fine here? It's still saying "changes requested" even tho I resolved everything.

**Review by zabi94 — APPROVED** [review]

**Comment by zabi94** [comment]
Don't worry, I was just too busy with work. I'll try it now

**Comment by zabi94** [comment]
Uploaded to curseforge and modrinth

**Pull request merged**
